offload/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -190,8 +190,8 @@ Error asyncMemCopy(bool UseMultipleSdmaEngines, void *Dst, hsa_agent_t DstAgent,
 #endif
 }
 
-Expected<std::string> getTargetTripleAndFeatures(hsa_agent_t Agent) {
-  std::string Target;
+Error getTargetTripleAndFeatures(hsa_agent_t Agent,
+                                 SmallVector<SmallString<32>> &Targets) {
```

Comment:
I don't see too much point of using `SmallString` here because every use of it is to convert it to `std::string`.

Reply:
`SmallString` is std::string with a template argument for the size of the SSO buffer. I would recommend we choose a number we expect to always fit the string so we don't need to malloc it.

Reply:
then 32 is sufficient here